### PR TITLE
EWL-3987: Feature/ewl 3987 refactor articles

### PR DESCRIPTION
**Pull request by rpama**
## Ticket(s)

**Jira Ticket**

- [EWL-3987: Refactor Topics Article Preview Related Patterns to use Pseudo-Patterns ](https://issues.ama-assn.org/browse/EWL-3987)

## Description

Removes previous article preview patterns and replaces with one template and json for several pseudo-patterns. Updates templates that contained this patterns to use the pseudo-patterns correctly. Makes some adjustments to styles for our vertical spacing issues, but this is kind of hinky.

## To Test

- [x] Molecules > Components > Article Preview (and variants)
- [x] Use pattern lab's show pattern info to track the other patterns that use these^^
- [x] See that nothing looks terribly broken
- [x] Review the css changes for any obvious badness

## Relevant Screenshots/GIFs

Use something like [Skitch](https://evernote.com/skitch/) or [GIPHY Capture](https://giphy.com/apps/giphycapture) to capture images/gifs to demonstrate behaviors.

## Remaining Tasks

- [ ] ~Need Robert to confirm with UX the element order for the video variant pattern.~
Since the direction was a change from what we had before, I've requested a separate ticket for that work from @froboy.

## Additional Notes

Anything more to add?

### CODE REVIEW

File: styleguide/source/_patterns/02-organisms/06-topic/01-topic-hero/_topic-hero.scss [hunk 1]
```diff
@@ -15,13 +15,12 @@
...
-  @include breakpoint($bp-med) {
-    @include child-top-gutters($gutter / 2);
-  }
-}
+// // Ensure top gutters for child elements.
+// .topic_hero .topic_article-preview {
+//   @include breakpoint($bp-med) {
+//     @include child-top-gutters($gutter / 2);
+//   }
+// }
```
Comment (labbydev) [inline styleguide/source/_patterns/02-organisms/06-topic/01-topic-hero/_topic-hero.scss]: This should be cleaned out if it isn't needed anymore

File: styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/_topic-topic-articles.scss [hunk 2]
```diff
@@ -19,9 +19,9 @@
 
 // Ensure consistent top margins for elements in this section.
 .topic_topic-articles .topic_article-preview > * {
-  @include gutter($margin-top-half...);
+  //@include gutter($margin-top-half...);
 }
```
Comment (labbydev) [inline styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/_topic-topic-articles.scss]: This should be cleaned out

File: styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/topic-topic-articles.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "video": {
+      "image": "",
+      "label": "true",
+      "metadata": "",
+      "related": false,
+      "title": "Topic Article Video Title",
+      "video": true
+    }
+  }
+}
```
Comment (labbydev) [inline styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/topic-topic-articles.json]: Line ending

File: styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/topic-topic-articles.twig [hunk 4]
```diff
@@ -1,4 +1,4 @@
 <div class="topic_topic-articles {{ class }}">
-    {% include 'molecules-article-preview' with { 'class': default } %}
-    {% include 'molecules-article-preview-video' with { 'class': default }  %}
+    {% include "molecules-article-preview" with {'topic_article' : topicTopicArticle.image } %}
+    {% include "molecules-article-preview" with {'topic_article' : topicTopicArticle.video } %}
 </div>
```
Comment (labbydev) [inline styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/topic-topic-articles.twig]: Line ending

File: styleguide/source/assets/css/style.scss [hunk 5]
```diff
@@ -209,7 +209,7 @@ So feel free to use any of these approaches. Or don't. It's totally up to you.
 @import "../../_patterns/02-organisms/06-topic/06-topic-promotion/_topic-promotion";
 @import "../../_patterns/01-molecules/08-lists/14-list-topic-related-articles/list-topic-related-articles";
 @import "../../_patterns/02-organisms/06-topic/08-topic-related-articles/topic-related-articles";
-@import "../../_patterns/01-molecules/06-components/22-article-preview/02-article-preview-related/article-preview-related";
+//@import "../../_patterns/01-molecules/06-components/22-article-preview/02-article-preview-related/article-preview-related";
```
Comment (labbydev) [inline styleguide/source/assets/css/style.scss]: Cleanup

File: styleguide/source/_patterns/02-organisms/06-topic/01-topic-hero/topic-hero.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    "second": {
+      "image": "",
+      "label": "true",
+      "metadata": "",
+      "related": false,
+      "title": "Article Video Title",
+      "video": true
+    }
+  }
+}
```
Comment (labbydev) [inline styleguide/source/_patterns/02-organisms/06-topic/01-topic-hero/topic-hero.json]: Line ending

File: styleguide/source/_patterns/01-molecules/08-lists/14-list-topic-related-articles/list-topic-related-articles.json [hunk 7]
```diff
@@ -2,13 +2,18 @@
...
     },
     "third": {
-      "title": "Lorem ipsum dolor sit amet, dictum dui, massa erat morbi facilisis"
+      "title": "Lorem ipsum dolor sit amet, dictum dui, massa erat morbi facilisis",
+      "related": true,
+      "image": "https://ipsumimage.appspot.com/140x100?l=7:5|140x100&amp;s=18"
     }
   }
-}
+}
```
Comment (labbydev) [inline styleguide/source/_patterns/01-molecules/08-lists/14-list-topic-related-articles/list-topic-related-articles.json]: Line ending

File: styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~video.json [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{
+  "topic_article": {
+    "image": "",
+    "label": "true",
+    "metadata": "",
+    "related": false,
+    "title": "Article Video Title",
+    "video": true
+  }
+}
```
Comment (labbydev) [inline styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~video.json]: Line ending

File: styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~related.json [hunk 9]
```diff
@@ -0,0 +1,9 @@
+{
+  "topic_article": {
+    "related": true,
+    "image": "https://ipsumimage.appspot.com/279x186?l=3:2|279x186&s=36",
+    "label": false,
+    "metadata": "",
+    "title": "Related Article Title"
+  }
+}
```
Comment (labbydev) [inline styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~related.json]: Line ending

File: styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~related-no-image.json [hunk 10]
```diff
@@ -0,0 +1,9 @@
+{
+  "topic_article": {
+    "related": true,
+    "image": "",
+    "label": false,
+    "metadata": "",
+    "title": "Related Article Title"
+  }
+}
```
Comment (labbydev) [inline styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~related-no-image.json]: Line ending

File: styleguide/source/_patterns/01-molecules/06-components/22-article-preview/_article-preview.scss [hunk 11]
```diff
@@ -1,25 +1,64 @@
...
+    @include grid__unit--cols(6);
+    order: 1;
+
+    @include breakpoint($bp-med) {
+      @include grid__unit--cols(9);
+      order: 2;
+      padding-right: 0;
+    }
+  }
+}
```
Comment (labbydev) [inline styleguide/source/_patterns/01-molecules/06-components/22-article-preview/_article-preview.scss]: Line ending

**Comment by rpama** [comment]
I have removed the commented out code. I was being overly cautious in the PR and then forgot to do clean-up. Whoops!

I have not, however, addressed the line-ending comments since I'm not entirely sure how we want to proceed with this. I'd like to get this PR merged (after @chelsiejohnston takes one more look for anything wonky with respect to the topics layouts) and then address the line-endings holistically. I've created spike ticket [EWL-4500](https://issues.ama-assn.org/browse/EWL-4005) for tackling the line ending question.

File: styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview.twig [hunk 12]
```diff
@@ -1,9 +1,29 @@
...
+      <div class="topic_article-preview_description">
+        {% include "atoms-paragraph" only %}
+      </div>
+    {% else %}
+      <div class="topic_article-preview_container-title"> {% include "molecules-link-black-h2" with { 'class': 'topic_article-preview_title', 'content': topic_article.title }%}</div>
+      {% if not topic_article.related %}
+        <div class="topic_article-preview_description">
+          {% include "atoms-paragraph" only %}
+        </div>
+        {% include "molecules-article-metadata" with { 'class': 'topic_article-preview_metadata' } %}
```
Comment (chelsiejohnston) [inline styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview.twig]: Suggestion to move this include outside of this conditional - this fixes the videos not having metadata and is a truer reflection of the content model

**Comment by rpama** [comment]
Updated pattern info in the `article-preview.md` file, but ticket EWL-3793 doesn't exist in Jira anymore so that was omitted.

**Comment by rpama** [comment]
@chelsiejohnston I believe I've now addressed all three of the issues you noted. The template's a little inelegant in its conditionals, but some of of that will go away once we get the ticket saying we don't need the weird title-position-swap for the video variant.

**Comment by chelsiejohnston** [comment]
LGTM, hooray! Thanks @rpama 

I know we were planning on having a conversation about this PR before we merged it, so I'm going to leave it here for now.

**Pull request merged**